fastai/vision/learner.py

```diff
@@ -114,6 +114,18 @@ def _cl_int_plot_top_losses(self, k, largest=True, figsize=(12,12)):
         cl = int(cl)
         im.show(ax=axes.flat[i], title=
             f'{classes[self.pred_class[idx]]}/{classes[cl]} / {self.losses[idx]:.2f} / {self.probs[idx][cl]:.2f}')
+        if(heatmap):
+            xb,_ = self.data.one_item(im)
+            xb = xb.cuda()
+            m = self.learn.model.eval()
+            with hook_output(m[0]) as hook_a:
+                with hook_output(m[0], grad= True) as hook_g:
```

Comment:
Are you using `hook_g`?

Reply:
I must confess that I did not understand code for heat-map in details. I just used Jeremy's code from class as it is. But now I looked at  Lesson 6: pets revisited, I started to understand use of hook_g. I think grad-cam is  more informative than heat-map.  When I used grad-cam with plot_loss (largest = True/Fasle), it gave information about why we got largest loss for some of the  validation images. 
plot_top_losses(largest=True)
<img width="1497" alt="interp plot_top_losses 9 figsize 15 11 largest true heatmap true" src="https://user-images.githubusercontent.com/9778099/52612203-5b8ffa80-2e4e-11e9-983d-d631e90774b0.png">
plot_top_losses(largest=False)
<img width="1497" alt="interp plot_top_losses 9 figsize 15 11 largest false heatmap true" src="https://user-images.githubusercontent.com/9778099/52612207-60ed4500-2e4e-11e9-84ef-237ed5eeded2.png">

Reply:
Ok, using grad Cam makes more sense, yes!